src/test/java/de/hpi/bpt/argos/api/EntityMappingEndpointTest.java

```diff
@@ -0,0 +1,355 @@
+		int entityMappings = getMappingsCount();
+
+		WebSocket webSocket = WebSocket.buildWebSocket();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createDeleteRequest(ARGOS_REST_HOST, getDeleteEntityMappingUri(testMapping.getId()));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		assertEquals(entityMappings - 1, getMappingsCount());
+		assertEquals(null, PersistenceAdapterImpl.getInstance().getEventEntityMapping(testMapping.getId()));
+		assertEquals(0, PersistenceAdapterImpl.getInstance().getMappingConditions(testMapping.getId()).size());
+
+		List<String> webSocketMessages = webSocket.awaitMessages(1, 1000);
+		ArgosTestUtil.assertWebSocketMessage(webSocketMessages.get(0), PersistenceArtifactUpdateType.DELETE, "EventEntityMapping");
+	}
+
+	@Test
+	public void testDeleteEntityMapping_InvalidId_BadRequest() {
+		EventEntityMapping testMapping = createMapping();
+		int entityMappings = getMappingsCount();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createDeleteRequest(ARGOS_REST_HOST, getDeleteEntityMappingUri(0));
+
+		assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
+		assertEquals(entityMappings, getMappingsCount());
+	}
+
+	@Test
+	public void testEditEntityMapping() throws Exception {
+		EventEntityMapping testMapping = createMapping();
+		EntityType newTargetEntityType = ArgosTestUtil.createEntityType(true);
+		List<TypeAttribute> newTargetEntityTypeAttributes = ArgosTestUtil.createEntityTypeAttributes(newTargetEntityType, true);
+		EventType newTargetEventType = ArgosTestUtil.createEventType(true, true);
+		List<TypeAttribute> newTargetEventTypeAttributes = ArgosTestUtil.createEventTypeAttributes(newTargetEventType, true);
+
+		String targetStatus = "TargetStatus_" + ArgosTestUtil.getCurrentTimestamp();
+		int entityMappings = getMappingsCount(newTargetEventType);
+
+		WebSocket webSocket = WebSocket.buildWebSocket();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createPutRequest(ARGOS_REST_HOST, getEditEntityMappingUri(testMapping.getId()));
+
+		JsonObject newMapping = new JsonObject();
+
+		newMapping.addProperty("EventTypeId", newTargetEventType.getId());
+		newMapping.addProperty("EntityTypeId", newTargetEntityType.getId());
+		newMapping.addProperty("TargetStatus", targetStatus);
+
+		JsonArray mappingConditions = new JsonArray();
+
+		JsonObject newMappingCondition = new JsonObject();
+		newMappingCondition.addProperty("EventTypeAttributeId", newTargetEventTypeAttributes.get(0).getId());
+		newMappingCondition.addProperty("EntityTypeAttributeId", newTargetEntityTypeAttributes.get(0).getId());
+
+		mappingConditions.add(newMappingCondition);
+
+		newMapping.add("EventEntityMappingConditions", mappingConditions);
+
+		request.setContent(serializer.toJson(newMapping));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		assertEquals(entityMappings + 1, getMappingsCount(newTargetEventType));
```

Comment:
Actually it should. Since I change the `EventTypeId` of the `Mapping`, there will be one more `Mapping` for the new `EventType` after the `Mapping` was edited.